src/ophyd_async/epics/pvi/pvi.py

```diff
@@ -296,3 +306,30 @@ async def fill_pvi_entries(
     # We call set name now the parent field has been set in all of the
     # introspect-initialized devices. This will recursively set the names.
     device.set_name(device.name)
+
+
+def pre_initialize_blocks(
+    device: Device, included_optional_fields: Optional[Tuple[str, ...]] = None
+):
+    """For intializing blocks at __init__ of ``device``."""
+    for name, device_type in get_type_hints(type(device)).items():
+        if name in ("_name", "parent"):
+            continue
+        device_type, is_optional = _strip_union(device_type)
+        if (
+            is_optional
+            and included_optional_fields
+            and name not in included_optional_fields
+        ):
```

Comment:
So we can make this:
```suggestion
        if is_optional and name not in included_optional_fields:
```